pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1195,12 +1234,24 @@ func (c *Controller) isPxImageBeingUpdated(toUpdate *corev1.StorageCluster) bool
 		(toUpdate.Spec.Version == "" || newVersion != toUpdate.Status.Version)
 }
 
-func getZoneMap(nodeList *v1.NodeList) map[string]uint64 {
+func getZoneMap(nodeList *v1.NodeList, filterLabelKey string, filterLabelValue string) map[string]uint64 {
 	cloudProviderName := getCloudProviderName(nodeList)
 	cloudProvider := cloudprovider.New(cloudProviderName)
 	zoneMap := map[string]uint64{}
 	for _, node := range nodeList.Items {
 		if zone, err := cloudProvider.GetZone(&node); err == nil {
+			if len(filterLabelKey) > 0 {
```

Comment:
is the error safe to ignore?

Reply:
Currently it's only if the node is nil, which won't be the case. I'll change it to handle the error  to protect against future changes to GetZone.